configstore/backends/docker_secret.py

```diff
@@ -9,6 +9,10 @@ def __init__(self, secrets_path=SECRETS_PATH):
         self.secrets_path = secrets_path
 
     def get_config(self, key):
+        path = os.path.join(self.secrets_path, key)
 
-        with open(os.path.join(self.secrets_path, key)) as fd:
+        if not os.path.exists(path):
+            return None
+
+        with open(path) as fd:
             return fd.readline().strip()
```

Comment:
This does two OS calls and has a race condition.  It’s probably minor for this package, but it doesn’t cost to follow the general rule (catch the ENOENT instead of checking exists).